src/components/media/audio/sound.tsx

```diff
@@ -0,0 +1,230 @@
+      }
+    }
+    function handleCanPlay() {
+      setErrorMessage('')
+      setIsLoading(false)
+    }
+    function handleError(errorMessage: string) {
+      setErrorMessage(errorMessage)
+    }
+
+    audioElement?.addEventListener('timeupdate', updateTime)
+    audioElement?.addEventListener('canplay', handleCanPlay)
+    audioElement?.addEventListener('error', () =>
+      handleError(loadingErrorMessage)
+    )
+    audioElement?.addEventListener('stalled', () =>
+      handleError(stalledErrorMessage)
+    )
+
+    return () => {
+      audioElement?.removeEventListener('timeupdate', updateTime)
+      audioElement?.removeEventListener('canplay', handleCanPlay)
+      audioElement?.removeEventListener('error', () =>
+        handleError(loadingErrorMessage)
+      )
+      audioElement?.removeEventListener('stalled', () =>
+        handleError(stalledErrorMessage)
+      )
+    }
+  }, [])
+
+  function renderSources() {
+    const invalidSrcErrorMessage = 'No valid audio sources were found'
+    const validSrcList = props.src.filter((src) => src.length > 0)
+
+    // Check if there are no valid sources
+    if (validSrcList.length === 0) {
+      setErrorMessage(invalidSrcErrorMessage)
+    } else {
+      return validSrcList.map((src, index) => <source key={index} src={src} />)
+    }
+  }
+
+  function renderTracks() {
+    if (props.captions && props.captions.length > 0) {
+      return props.captions.map((captionSrc, index) => (
+        <track key={index} src={captionSrc} kind="captions" default />
+      ))
+    }
+  }
+
+  if (errorMessage.length > 0) {
+    return (
+      <Alert severity="error" data-cy="error">
+        {errorMessage}
+      </Alert>
+    )
+  }
+
+  function renderTitle() {
+    return (
+      <Typography
+        variant="h6"
+        className="rustic-sound-title"
+        data-cy="sound-player-title"
+      >
+        {props.title}
+      </Typography>
+    )
+  }
+
+  function renderCaptionsToggle() {
+    if (props.captions && props.captions.length > 0) {
+      return (
+        <IconButton
+          onClick={() => setAreCaptionsShown(!areCaptionsShown)}
+          aria-label={areCaptionsShown ? 'Hide captions' : 'Show captions'}
+          data-cy="captions-toggle"
+        >
+          {areCaptionsShown ? (
+            <ClosedCaptionRoundedIcon color="primary" />
+          ) : (
+            <ClosedCaptionDisabledRoundedIcon color="primary" />
+          )}
+        </IconButton>
+      )
+    }
+  }
+
+  return (
+    <Box className="rustic-sound" data-cy="audio">
+      <video
+        data-cy="audio-element"
+        ref={audioRef}
+        className={areCaptionsShown ? 'rustic-has-captions' : ''}
+      >
+        {renderSources()}
+        {areCaptionsShown && renderTracks()}
+      </video>
+
+      {isLoading || !audioElement ? (
+        <CircularProgress data-cy="spinner" />
+      ) : (
+        <>
+          {!isMobile && renderTitle()}
```

Comment:
I think it will be better to have a two functions which render for mobile and desktop. This will make it easier to visualize and manage for future updates.